packages/flutter_tools/lib/src/android/build_validation.dart

```diff
@@ -11,6 +11,13 @@ const String kSupportedAbis = 'https://flutter.dev/docs/deployment/android#what-
 /// Validates that the build mode and build number are valid for a given build.
 void validateBuild(AndroidBuildInfo androidBuildInfo) {
   final BuildInfo buildInfo = androidBuildInfo.buildInfo;
+  if (buildInfo.codeSizeDirectory != null && androidBuildInfo.targetArchs.length > 1) {
+    throwToolExit(
+      'Cannot perform code size analysis when building for multiple ABIs. '
```

Comment:
oh cool, I like this solution. But can we also edit the help for the --analyze-size flag then to say you need to also specify this? Otherwise, all users will always run `flutter build apk --analyze-size` one time and always get a warning to re-run it a second time.

Reply:
Yeah, will update the analyze-size flag